connect-inject/container_init_test.go

```diff
@@ -70,6 +70,28 @@ func TestHandlerContainerInit(t *testing.T) {
 			"",
 		},
 
+		{
+			"Upstream datacenter specified",
+			func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations[annotationService] = "web"
+				pod.Annotations[annotationUpstreams] = "db:1234:dc1"
+				return pod
+			},
+			`datacenter = "dc1"`,
+			"",
+		},
+
+		{
+			"No Upstream datacenter specified",
+			func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations[annotationService] = "web"
+				pod.Annotations[annotationUpstreams] = "db:1234"
+				return pod
+			},
+			"",
+			`datacenter = "dc1"`,
```

Comment:
In this case, it would be good to make sure that no datacenter info is included rather than making sure a specific value is not present.
```suggestion
			`datacenter`,
```